src/picasso-definition/components/navigation-panel/index.js

```diff
@@ -0,0 +1,101 @@
+import NUMBERS from '../../../constants/numbers';
+import KEYS from '../../../constants/keys';
+
+// Translation (up/down/left/right)
+function translate({ viewHandler, direction, percent }) {
+  const { xAxisMax, xAxisMin, yAxisMax, yAxisMin } = viewHandler.getDataView();
+  let deltaX = 0;
+  let deltaY = 0;
+  if (direction === 'x') {
+    deltaX = ((xAxisMax - xAxisMin) * percent) / 100;
+  } else {
+    deltaY = ((yAxisMax - yAxisMin) * percent) / 100;
+  }
+  const dataView = {
+    xAxisMin: xAxisMin + deltaX,
+    xAxisMax: xAxisMax + deltaX,
+    yAxisMax: yAxisMax + deltaY,
+    yAxisMin: yAxisMin + deltaY,
+  };
+  viewHandler.setDataView(dataView);
+}
+
+// Zoom in/ Zoom out
+function transform(scale, start, end, factor) {
```

Comment:
transform and below zoom function are quite similar to the functions in `src/utils/math/zoom.js`, is that possible to modify use them instead?

Reply:
good idea!